### PR TITLE
[8.0] Added configurable chunksize option to dirac_dms_replicate_and_r…

### PR DESCRIPTION
**Pull request by marianne013**
…egister_request script

BEGINRELEASENOTES

*DataManagement
CHANGE: dirac-dms-replicate-and-register-request: Make request chunk size configurable; default behaviour unchanged.

ENDRELEASENOTES

### CODE REVIEW

File: src/DIRAC/DataManagementSystem/scripts/dirac_dms_replicate_and_register_request.py [hunk 1]
```diff
@@ -22,15 +22,19 @@ def getLFNList(arg):
...
     Script.registerArgument(["targetSE:     target SE"])
     Script.parseCommandLine()
+
+    chnksize = 100
     for switch in Script.getUnprocessedSwitches():
         if switch[0] == "C" or switch[0].lower() == "catalog":
             catalog = switch[1]
-
+        if switch[0] == "N" or switch[0].lower() == "chunksize":
+            chnksize = int(switch[1])
```
Comment (andresailer) [inline src/DIRAC/DataManagementSystem/scripts/dirac_dms_replicate_and_register_request.py]:
```suggestion
            chunksize = int(switch[1])
```

File: src/DIRAC/DataManagementSystem/scripts/dirac_dms_replicate_and_register_request.py [hunk 2]
```diff
@@ -54,7 +58,7 @@ def main():
     from DIRAC.Resources.Catalog.FileCatalog import FileCatalog
     from DIRAC.Core.Utilities.List import breakListIntoChunks
 
-    lfnChunks = breakListIntoChunks(lfnList, 100)
+    lfnChunks = breakListIntoChunks(lfnList, chnksize)
```
Comment (andresailer) [inline src/DIRAC/DataManagementSystem/scripts/dirac_dms_replicate_and_register_request.py]:
```suggestion
    lfnChunks = breakListIntoChunks(lfnList, chunksize)
```

File: src/DIRAC/DataManagementSystem/scripts/dirac_dms_replicate_and_register_request.py [hunk 3]
```diff
@@ -22,15 +22,19 @@ def getLFNList(arg):
...
     catalog = None
     Script.registerSwitch("C:", "Catalog=", "Catalog to use")
+    Script.registerSwitch("N:", "ChunkSize=", "Number of files per request")
     # Registering arguments will automatically add their description to the help menu
     Script.registerArgument(" requestName:  a request name")
     Script.registerArgument(" LFNs:         single LFN or file with LFNs")
     Script.registerArgument(["targetSE:     target SE"])
     Script.parseCommandLine()
+
+    chnksize = 100
```
Comment (andresailer) [inline src/DIRAC/DataManagementSystem/scripts/dirac_dms_replicate_and_register_request.py]:
```suggestion
    chunksize = 100
```

**Review by andresailer — COMMENTED** [review]
We can afford the extra 'u's

**Comment by marianne013** [comment]
I left the 'u' out to make it clear that chunksize (line 36) is not the same as chnksize (line 37).  At least not in programming terms. I don't do much programming, I need all the help I can get.
#featurenotbug

**Comment by andresailer** [comment]
One is a `"string"` and the other a `variable`, no harm to use the identical characters.
The same is done for `catalog`. It is much harder to read misspelled text, and remember which misspelling was used.

**Comment by marianne013** [comment]
The "catalog" is not mine. 
```<digs heels in>```
Took me some time to work out how to display the brackets.

**Comment by andresailer** [comment]
Yeah, I know, you should do as the romans with the catalog.

**Comment by marianne013** [comment]
I still think my way is better, and I can't even hit "commit suggestion" as it will then fail the "commit message formatting" test.

File: src/DIRAC/DataManagementSystem/scripts/dirac_dms_replicate_and_register_request.py [hunk 4]
```diff
@@ -22,15 +22,19 @@ def getLFNList(arg):
 def main():
     catalog = None
     Script.registerSwitch("C:", "Catalog=", "Catalog to use")
+    Script.registerSwitch("N:", "ChunkSize=", "Number of files per request")
```
Comment (andresailer) [inline src/DIRAC/DataManagementSystem/scripts/dirac_dms_replicate_and_register_request.py:25]:
```suggestion
    Script.registerSwitch("N:", "ChnkSze=", "Number of files per request")
```

To not confuse it with line 36 and 37!

Edit: /s

**Comment by andresailer** [comment]
> I can't even hit "commit suggestion" as it will then fail the "commit message formatting" test.

You can "add suggestion to batch" and then type your own commit message.
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request

**Comment by marianne013** [comment]
I've added various 'u'. Everyone happy ?

**Review by fstagni — APPROVED** [review]

**Pull request merged**